### ISSUE TITLE
UnixDatagram does not support vectored send

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust#68612]
The `UnixDatagram` type provides `send` and `send_to` methods, but does not provide a vectored send operation. Such an operation is particularly helpful for datagrams, where separate send operations result in separate datagrams; a vectored send allows sending data from multiple buffers in a single datagram.

Given that `UnixDatagram` only runs on UNIX, where `writev` works just fine on a socket, such an operation could use `writev`; alternatively, this could use `sendmsg`, which supports supplying an iovec. `sendmsg` would also allow a vectored `send_to` operation.

**Comment by sfackler** [comment]
We don't support vectored send/recv on UdpSocket either currently - PRs welcome for both!

**Comment by joshtriplett** [comment]
@sfackler Makes sense; `UdpSocket` filed separately as https://github.com/rust-lang/rust/issues/68617.

**Comment by contrun** [comment]
I have just added vectored send for `send`. It seems that we don't have binding for `sendmsg` from libc.
